foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_11_0/DockerModulePomDependencyTypeMigration.java

```diff
@@ -0,0 +1,110 @@
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractPomMigration;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.technologybrewery.baton.util.pom.PomHelper;
+import org.technologybrewery.baton.util.pom.PomModifications;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.END;
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.START;
+
+/**
+ * This migration fixes the build cache dependency check by removing the dependency `pom` type for the java pipeline
+ * and replace the dependency type with `habushu` if it's python pipeline
+ */
+public class DockerModulePomDependencyTypeMigration extends AbstractPomMigration {
+    private static final String PIPELINE_GROUP_ID = "${project.groupId}";
+    private static final String JAVA_PIPELINE = "java";
+    private static final String PYTHON_PIPELINE = "python";
+    private static final String HABUSHU = "habushu";
+    private static final String POM = "pom";
+
+    @Override
+    protected boolean shouldExecuteOnFile(File pomFile) {
+        Model model = PomHelper.getLocationAnnotatedModel(pomFile);
+        if (model.getPackaging().equals("docker-build")) {
+            // find the pipeline artifacts
+            Map<String, String> pipelineArtifactIds = getPipelineArtifactIds(pomFile);
+            return getDependenciesNeedToBeUpdated(model, pipelineArtifactIds).size() > 0;
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean performMigration(File pomFile) {
+        Model model = PomHelper.getLocationAnnotatedModel(pomFile);
+
+        PomModifications modifications;
+        Map<String, String> pipelineArtifactIds = getPipelineArtifactIds(pomFile);
+        List<Dependency> dependenciesToBeUpdated = getDependenciesNeedToBeUpdated(model, pipelineArtifactIds);
+
+        modifications = new PomModifications();
+        for (Dependency dependency: dependenciesToBeUpdated) {
+
+            if (POM.equals(dependency.getType())) {
+                if (pipelineArtifactIds.get(dependency.getArtifactId()).equals(JAVA_PIPELINE)) {
+                    // delete the dependency type if it's java pipeline
+                    modifications.add(deleteTag(dependency, "type"));
+                } else {
+                    // replace the dependency type if it's python pipeline
+                    modifications.add(replaceInTag(dependency, "type", HABUSHU));
+                }
+            } else if (pipelineArtifactIds.get(dependency.getArtifactId()).equals(PYTHON_PIPELINE)) {
```

Comment:
Q: Is this condition possible? If it's a python pipeline, then it will be `habushu` packaged. Therefore the only two dependency types that would be valid are `pom` and `habushu`, right? If it weren't one of those two, I'd expect their build to be failing.